### PR TITLE
Update GitHub actions and Python Dependencies

### PR DESCRIPTION
**Pull request by jmgrady**
This PR incorporates a number of maintenance updates to the Python code and the GitHub actions:

- Updates the Python dependencies for development, the `maintenance` container, and the `deploy` container;
- Updates versions of actions that are used in _The Combine_ workflows:
   - `action/checkout` to 3.3.0
   - `actions/upload-artifact` to 3.1.2
   - `actions/setup-node` to 3.6.0
   - `docker/build-push-action` to 3.3.0
   - `ossf/scorecard-action` to 2.1.2
- Changes the `egress-policy` to `audit` for workflow steps that build the backend.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/TheCombine/1821)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # User guide build environment.
-FROM python:3.10 AS user_guide_builder
+FROM python:3.11 AS user_guide_builder
```
[the remark this answers is not in the document]
Comment (jmgrady) [reply Dockerfile]: Done

File: Backend/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Docker multi-stage build.
-FROM mcr.microsoft.com/dotnet/sdk:6.0 AS builder
+FROM bitnami/dotnet-sdk:6 AS builder
```
[the remark this answers is not in the document]
Comment (jmgrady) [reply Backend/Dockerfile]: Done

File: Backend/Dockerfile [hunk 3]
```diff
@@ -11,7 +11,7 @@
 RUN dotnet publish -c Release -o build
 
 # Build runtime image.
-FROM mcr.microsoft.com/dotnet/aspnet:6.0
+FROM bitnami/aspnet-core:6
```
[the remark this answers is not in the document]
Comment (jmgrady) [reply Backend/Dockerfile]: Done

**Review by jasonleenaylor — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, 9 of 14 files at r2, 2 of 3 files at r3, 2 of 3 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/TheCombine/1821#-:-NMWIE7J7Jokhpe7jdHJ:bnfp4nl)* status: all files reviewed, 3 unresolved discussions (waiting on @github-code-scanning[bot])

**Pull request merged**